src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
```

Comment:
PullProofAI

The `JwtStrategy` class constructor and setup look good. Ensure that the `Env.JWT.AUTH_TOKEN` is properly configured and secure.



<!-- This is an auto-generated comment by PullProofAI -->